test/gate_test.dart

```diff
@@ -430,31 +430,72 @@ void main() {
     });
 
-    test('Index Logic by Logic test', () async {
+    test('Index Logic(8bit) by Logic test', () async {
       final gtm = IndexGateTestModule(Logic(width: 8), Logic(width: 8));
       await gtm.build();
       final vectors = [
         Vector({'original': 14, 'index': 0}, {'index_output': 0}),
         Vector({'original': 14, 'index': 2}, {'index_output': 1}),
         Vector({'original': 14, 'index': LogicValue.x},
-            {'index_output': LogicValue.x}),
+            {'index_output': LogicValue.x})
+      ];
+      await SimCompare.checkFunctionalVector(gtm, vectors);
+      final simResult = SimCompare.iverilogVector(
+          gtm.generateSynth(), gtm.runtimeType.toString(), vectors,
+          signalToWidthMap: {'original': 8, 'index': 8});
+      expect(simResult, equals(true));
+    });
+
+    test('Index Logic(1bit) by Logic test', () async {
+      final gtm = IndexGateTestModule(Logic(), Logic(width: 8));
+      await gtm.build();
+      final vectors = [
         Vector({'original': LogicValue.x, 'index': LogicValue.x},
             {'index_output': LogicValue.x}),
         Vector({'original': LogicValue.x, 'index': 0},
-            {'index_output': LogicValue.x})
+            {'index_output': LogicValue.x}),
+        Vector({'original': LogicValue.x, 'index': 1},
+            {'index_output': LogicValue.x}),
+        Vector({'original': LogicValue.x, 'index': 2},
+            {'index_output': LogicValue.x}),
+        Vector({'original': LogicValue.one, 'index': 0},
+            {'index_output': LogicValue.one}),
+        Vector({'original': LogicValue.one, 'index': 1},
+            {'index_output': LogicValue.one}),
+        Vector({'original': LogicValue.one, 'index': 2},
+            {'index_output': LogicValue.one}),
+        Vector({'original': LogicValue.zero, 'index': 0},
+            {'index_output': LogicValue.zero}),
+        Vector({'original': LogicValue.zero, 'index': 1},
+            {'index_output': LogicValue.zero}),
+        Vector({'original': LogicValue.zero, 'index': 2},
+            {'index_output': LogicValue.zero})
       ];
       await SimCompare.checkFunctionalVector(gtm, vectors);
       final simResult = SimCompare.iverilogVector(
           gtm.generateSynth(), gtm.runtimeType.toString(), vectors,
-          signalToWidthMap: {'original': 8, 'index': 8});
+          signalToWidthMap: {'original': 1, 'index': 8});
       expect(simResult, equals(true));
     });
 
     test('Index Logic by an Integer test', () {
       final testLogic = Logic(width: 8)..put(14);
+      final testLogicOne = Logic()..put(LogicValue.one);
+      final testLogicZero = Logic()..put(LogicValue.zero);
+      final testLogicInvalid = Logic()..put(LogicValue.x);
+
       expect(testLogic[0].value.toInt(), 0);
       expect(testLogic[2].value.toInt(), 1);
       expect(() => testLogic[10], throwsException);
+
+      expect(testLogicOne[0].value.toInt(), 1);
+      expect(testLogicOne[1].value.toInt(), 1);
```

Comment:
1-bit accesses with out of range index should be exceptions, which should happen with other changes mentioned